log4j-core/src/main/java/org/apache/logging/log4j/core/appender/rolling/PatternProcessor.java

```diff
@@ -343,6 +347,15 @@ private RolloverFrequency calculateFrequency(final String pattern) {
         return null;
     }
 
+    private static TimeZone calculateTimeZone(DatePatternConverter dateConverter) {
+        TimeZone timeZone = dateConverter.getTimeZone();
+        if (timeZone != null) {
+            return timeZone;
+        } else {
+            return Calendar.getInstance().getTimeZone();
+        }
+    }
```

Comment:
If `getTimeZone()` didn't return `null`, this would be unnecessary.

Reply:
Ok.